homeassistant/components/fluss/__init__.py

```diff
@@ -0,0 +1,43 @@
+"""The Fluss+ integration."""
+
+from __future__ import annotations
+
+import logging
+
+from fluss_api import (
+    FlussApiClient,
+    FlussApiClientAuthenticationError,
+    FlussApiClientCommunicationError,
+    FlussApiClientError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+LOGGER = logging.getLogger(__package__)
+
+PLATFORMS: list[Platform] = [Platform.BUTTON]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Fluss+ from a config entry."""
+
+    try:
+        api = FlussApiClient(entry.data[CONF_API_KEY])
+    except FlussApiClientAuthenticationError as e:
+        LOGGER.error("Authentication error initializing FlussApiClient: %s", e)
+        raise ConfigEntryAuthFailed from e
+    except FlussApiClientCommunicationError as e:
+        LOGGER.error("Communication error initializing FlussApiClient: %s", e)
+        raise ConfigEntryNotReady from e
+    except FlussApiClientError as e:
+        LOGGER.error("General error initializing FlussApiClient: %s", e)
+        raise ConfigEntryNotReady from e
```

Comment:
```suggestion
    try:
        api = FlussApiClient(entry.data[CONF_API_KEY])
    except FlussApiClientAuthenticationError as e:
        raise ConfigEntryAuthFailed from e
    except (FlussApiClientCommunicationError, FlussApiClientError) as e:
        raise ConfigEntryNotReady from e
```
will be logged automatically